File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_3/problem_3.py
def reverse(a, s, e):
    if s >= e:
        return a
    x = a[s]
    a[s] = a[e]
    a[e] = x
    
    return reverse(a, s + 1, e - 1)

def main():
    nums = [1.2, 5.3, 9.12, 24.5]
    s = 0
    e = len(nums) - 1

    print(f"Reverse of {nums} : {reverse(nums, s, e)}")


if __name__ == "__main__":
    main()

File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem3/Problem3.java
import java.util.Arrays;

public class Problem3 {

    public static double[] reverse(double[] a, int s, int e){
        if (s >= e) {
            return a;
        }
        double x = a[s];
        a[s] = a[e];
        a[e] = x ;

        return reverse(a, s + 1, e - 1);

    }

    public static void main(String[] args) {
        double[] nums = {1.2, 5.3, 9.12, 24.5};
        int s = 0;
        int e = nums.length - 1;

        System.out.println("Reverse of " + Arrays.toString(nums) + " : " + Arrays.toString(reverse(nums, s, e)));
    }
}
